restorer.go

```diff
@@ -93,16 +101,47 @@ func (r *Restorer) Restore(cache Cache) error {
 			}
 		}
 	}
+
+	if api.MustParse(r.Platform.API()).AtLeast("0.8") {
+		if r.LayersMetadata.BOM != nil && r.LayersMetadata.BOM.SHA != "" {
+			g.Go(func() error {
+				return r.restorePreviousLayer(r.LayersMetadata.BOM.SHA)
```

Comment:
It's kind of weird that `PreviousImage` lives on the restorer whereas `cache` is passed in. Maybe they should both be passed in (because they are both sources of things to restore)?

Reply:
_Fixed in latest commit._